core/database/src/main/java/com/mifos/room/entities/templates/loans/LoanTemplate.kt

```diff
@@ -44,115 +44,115 @@ import kotlinx.parcelize.Parcelize
+    val repaymentEvery: Int? = null,
 
-    var repaymentFrequencyType: RepaymentFrequencyType? = null,
+    val repaymentFrequencyType: RepaymentFrequencyType? = null,
 
-    var interestRatePerPeriod: Double? = null,
+    val interestRatePerPeriod: Double? = null,
 
-    var interestRateFrequencyType: InterestRateFrequencyType? = null,
+    val interestRateFrequencyType: InterestRateFrequencyType? = null,
 
-    var annualInterestRate: Double? = null,
+    val annualInterestRate: Double? = null,
 
-    var isFloatingInterestRate: Boolean? = null,
+    val isFloatingInterestRate: Boolean? = null,
 
-    var amortizationType: AmortizationType? = null,
+    val amortizationType: AmortizationType? = null,
 
-    var interestType: InterestType? = null,
+    val interestType: InterestType? = null,
 
-    var interestCalculationPeriodType: InterestCalculationPeriodType? = null,
+    val interestCalculationPeriodType: InterestCalculationPeriodType? = null,
 
-    var allowPartialPeriodInterestCalcualtion: Boolean? = null,
+    val allowPartialPeriodInterestCalcualtion: Boolean? = null,
 
-    var transactionProcessingStrategyId: Int? = null,
+    val transactionProcessingStrategyId: Int? = null,
 
-    var graceOnArrearsAgeing: Int? = null,
+    val graceOnArrearsAgeing: Int? = null,
 
-    var timeline: Timeline? = null,
+    val timeline: Timeline? = null,
 
-    var productOptions: List<ProductOptions> = ArrayList(),
+    val productOptions: List<ProductOptions> = ArrayList(),
 
-    var dataTables: ArrayList<DataTable> = ArrayList(),
+    val dataTables: ArrayList<DataTable> = ArrayList(),
 
-    var loanOfficerOptions: List<LoanOfficerOptions> = ArrayList(),
+    val loanOfficerOptions: List<LoanOfficerOptions> = ArrayList(),
 
-    var loanPurposeOptions: List<LoanPurposeOptions> = ArrayList(),
+    val loanPurposeOptions: List<LoanPurposeOptions> = ArrayList(),
 
-    var fundOptions: List<FundOptions> = ArrayList(),
+    val fundOptions: List<FundOptions> = ArrayList(),
 
-    var termFrequencyTypeOptions: List<TermFrequencyTypeOptions> = ArrayList(),
+    val termFrequencyTypeOptions: List<TermFrequencyTypeOptions> = ArrayList(),
 
-    var repaymentFrequencyTypeOptions: List<RepaymentFrequencyTypeOptions> = ArrayList(),
+    val repaymentFrequencyTypeOptions: List<RepaymentFrequencyTypeOptions> = ArrayList(),
 
-    var repaymentFrequencyNthDayTypeOptions: List<RepaymentFrequencyNthDayTypeOptions> = ArrayList(),
+    val repaymentFrequencyNthDayTypeOptions: List<RepaymentFrequencyNthDayTypeOptions> = ArrayList(),
 
-    var repaymentFrequencyDaysOfWeekTypeOptions: List<RepaymentFrequencyDaysOfWeekTypeOptions> = ArrayList(),
+    val repaymentFrequencyDaysOfWeekTypeOptions: List<RepaymentFrequencyDaysOfWeekTypeOptions> = ArrayList(),
 
-    var interestRateFrequencyTypeOptions: List<InterestRateFrequencyTypeOptions> = ArrayList(),
+    val interestRateFrequencyTypeOptions: List<InterestRateFrequencyTypeOptions> = ArrayList(),
```

Comment:
Please use emptyList(), emptySet(), or emptyMap() instead of ArrayList(), HashSet(), or other explicit collection initializations in the database module. 
By minimizing pointless object creation, enforcing immutability to prevent accidental changes, and adhering to Kotlin best practices for cleaner, more maintainable code, using these functions increases efficiency.

Apply this to all collections (Lists, Sets, Maps) across the module where applicable. Let’s keep our code efficient and clean!